source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/AdaptiveCardsSharedModelUnitTest.cpp

```diff
@@ -0,0 +1,197 @@
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z,SHORT)}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{DATE(2017-02-13T20:46:30Z,SHORT)}", blck.GetText());
+        }
+
+        TEST_METHOD(ISO8601TransformBadInputTestBadYear)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017a02-13T20:46:30Z,SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{DATE(2017a02-13T20:46:30Z,SHORT)}}", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToDateTest1)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z,SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("02/13/17", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToTimeTest2)
+        {
+            TextBlock blck;
+            string testString = "{{TIME(2017-10-27T18:19:09Z)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("11:19:09 AM", blck.GetText());
+        }
+        TEST_METHOD(TransformToDateTest3)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-10-28T01:47:00Z,SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("10/27/17", blck.GetText());
+        }
+        TEST_METHOD(TransformToTimeTest)
+        {
+            TextBlock blck;
+            string testString = "{{TIME(2017-10-28T02:17:00Z)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("07:17:00 PM", blck.GetText());
+        }
+        TEST_METHOD(TransformToDateTest4)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-10-28T02:17:00Z,SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("10/27/17", blck.GetText());
+        }
+        TEST_METHOD(TransformToTimeTest1)
+        {
+            TextBlock blck;
+            // paris
+            string testString = "{{TIME(2017-10-28T04:20:00+02:00)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("07:20:00 PM", blck.GetText());
+        }
+        TEST_METHOD(TransformToDateTest5)
+        {
+            TextBlock blck;
+            // paris
+            string testString = "{{DATE(2017-10-28T04:20:00+02:00)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("10/27/17", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToTimeTest3)
+        {
+            TextBlock blck;
+            // seoul 
+            string testString = "{{TIME(2017-10-28T11:25:00+09:00)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("07:25:00 PM", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToDateTest6)
+        {
+            TextBlock blck;
+            // seoul 
+            string testString = "{{DATE(2017-10-28T11:25:00+09:00,LONG)}}";
```

Comment:
>+09:00 [](start = 59, length = 6)

Please check with Matt how the timezone should be specified. The Typescript renderer uses +09 and breaks on 09:00

Reply:
o.k.

---
In reply to: [149213501](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149213501) [](ancestors = 149213501)